### PR TITLE
scylla-sstable: handle attempt to load schema for non-existent tables more gracefully

### PR DESCRIPTION
**Pull request by denesb**
In other words, print more user-friendly messages, and avoid crashing.
Specifically:
* Don't crash when attempting to load schema tables from configured data-dir, while configuration does not have any configured data-directories.
* Detect the case where schema mutations have no rows for the current table -- the keyspace exists, but the table doesn't.
* Add negative tests for schema-loading.

Fixes: https://github.com/scylladb/scylladb/issues/16459

### CODE REVIEW

File: tools/schema_loader.cc [hunk 1]
```diff
@@ -41,6 +41,8 @@
 
 namespace {
 
+logging::logger sllog("schema_loader");
```
Comment (nyh) [inline tools/schema_loader.cc:44]: Although logging is better than nothing, so I am not against this patch - I don't think "normal" users will want to use it, and will prefer regular error messages saying exactly what's wrong - not intermixed with a dozen other messages about what went right.

File: test/cql-pytest/test_tools.py [hunk 2]
```diff
@@ -586,6 +586,15 @@ def check(self, scylla_path, extra_args, sstable, dump_reference, cwd=None, env=
         dump = list(dump.values())[0]
         assert dump == dump_reference
 
+    def check_fail(self, scylla_path, extra_args, sstable, error_msg=None):
```
Comment (nyh) [inline test/cql-pytest/test_tools.py:589]: nitpick: the name "check_fail" is too general. It test a very specific tool and very specific usage of it - not some general failure.

**Review by nyh — APPROVED** [review]
Looks good. I wrote some trivial nitpicks, but none of it is essential.

File: test/cql-pytest/test_tools.py (hunk 2 above)
Comment (denesb) [reply test/cql-pytest/test_tools.py:589]: This is a member of a class, which provides context. Note that the above method is also just called `check()`.

File: tools/schema_loader.cc (hunk 1 above)
Comment (denesb) [reply tools/schema_loader.cc:44]: A agree with you. The logging here is for tests and developers diagnosing the bug. By default, none of these logs are shown, they are all debug and trace.
The next patch is making sure we get a user-readable error in a case we got a cryptic message before.

**Pull request merged**